account/local_account.go

```diff
@@ -13,41 +14,101 @@ import (
 }
 
-// NewLocalAccount creates a new LocalAccount
-func NewLocalAccount() (Account, error) {
+// NewLocalAccount creates a new local account with a freshly generated private key
+func NewLocalAccount(name string) (*LocalAccount, error) {
 	k, err := key.NewSoft()
 	if err != nil {
 		return nil, err
 	}
 	return &LocalAccount{
-		SoftKey: k,
+		name:    name,
+		softKey: k,
 	}, nil
 }
 
-func Import(keyPath string) (Account, error) {
-	k, err := key.LoadSoft(keyPath)
+// NewLocalAccountFromPrivateKey imports a local account from a hex-encoded private key string
+func NewLocalAccountFromPrivateKey(name string, privateKeyHex string) (*LocalAccount, error) {
+	k, err := key.LoadSoftFromBytes([]byte(privateKeyHex))
 	if err != nil {
 		return nil, err
 	}
 	return &LocalAccount{
-		SoftKey: k,
+		name:    name,
+		softKey: k,
 	}, nil
 }
 
+// NewLocalAccountFromKeyPath imports a local account from a file path containing a private key
+func NewLocalAccountFromKeyPath(name string, keyPath string) (*LocalAccount, error) {
+	kb, err := os.ReadFile(keyPath)
+	if err != nil {
+		return nil, err
+	}
+	return NewLocalAccountFromPrivateKey(name, string(kb))
+}
+
+func (a *LocalAccount) Name() string {
+	return a.name
+}
+
 func (a *LocalAccount) GetPChainAddress(network network.Network) (string, error) {
-	if a.SoftKey == nil {
-		return "", fmt.Errorf("SoftKey not initialized")
+	if a.softKey == nil {
+		return "", fmt.Errorf("uninitialized local account")
 	}
-	pchainAddrs, err := a.SoftKey.GetNetworkChainAddress(network, "P")
-	return pchainAddrs[0], err
+	pchainAddrs, err := a.softKey.GetNetworkChainAddress(network, "P")
+	if err != nil {
+		return "", err
+	}
+	if len(pchainAddrs) == 0 {
+		return "", fmt.Errorf("no P-Chain address found")
+	}
+	return pchainAddrs[0], nil
+}
+
+func (a *LocalAccount) GetXChainAddress(network network.Network) (string, error) {
+	if a.softKey == nil {
+		return "", fmt.Errorf("uninitialized local account")
+	}
+	xchainAddrs, err := a.softKey.GetNetworkChainAddress(network, "X")
+	if err != nil {
+		return "", err
+	}
+	if len(xchainAddrs) == 0 {
+		return "", fmt.Errorf("no X-Chain address found")
+	}
+	return xchainAddrs[0], nil
+}
+
+func (a *LocalAccount) GetCChainAddress() (string, error) {
+	if a.softKey == nil {
+		return "", fmt.Errorf("uninitialized local account")
+	}
+	// C-Chain uses EVM address format (0x...)
+	return a.softKey.C(), nil
+}
+
+func (a *LocalAccount) GetEVMAddress() (string, error) {
+	if a.softKey == nil {
+		return "", fmt.Errorf("uninitialized local account")
+	}
+	return a.softKey.C(), nil
 }
 
 func (a *LocalAccount) GetKeychain() (*secp256k1fx.Keychain, error) {
-	if a.SoftKey == nil {
-		return nil, fmt.Errorf("SoftKey not initialized")
+	if a.softKey == nil {
+		return nil, fmt.Errorf("uninitialized local account")
+	}
+	return a.softKey.KeyChain(), nil
+}
+
+// PrivateKey exports the private key in hex format
+func (a *LocalAccount) PrivateKey() (string, error) {
```

Comment:
Is the intention for users to use LocalAccount as Account in their code?
Cuz if so, I would make LocalAccount methods set 1 to 1 with Account interface (either removing the method or adding it to the Account interface).